diff -uNr xf86-input-keyboard-clean/configure xf86-input-keyboard-workdir/configure
--- xf86-input-keyboard-clean/configure	2022-07-11 01:28:12.000000000 +0200
+++ xf86-input-keyboard-workdir/configure	2025-09-06 17:01:12.012094097 +0200
@@ -19315,29 +19315,7 @@
 
 
 # The keyboard driver code is O/S specific
-case $host_os in
-  linux*)
-    as_fn_error $? "This is not the keyboard driver you are looking for. Use evdev or libinput." "$LINENO" 5
-    ;;
-
-  freebsd* | kfreebsd-gnu* | netbsd* | openbsd* | dragonfly*)
-    IS_BSD="yes"
-    ;;
-
-  solaris*)
-    IS_SOLARIS="yes"
-    ;;
-
-  gnu*)
-    IS_HURD="yes"
-    ;;
 
-  *)
-    as_fn_error $? "Your operating system is not supported by the kbd driver.
-                  Contact xorg@lists.freedesktop.org if you are interested in
-                  porting it." "$LINENO" 5
-    ;;
-esac
 case $host_os in
   netbsd*)
     case $host in
