diff -urN --no-dereference mlibc-clean/meson.build mlibc-workdir/meson.build
--- mlibc-clean/meson.build
+++ mlibc-workdir/meson.build
@@ -265,6 +265,10 @@
 	internal_conf.set10('MLIBC_MAP_DSO_SEGMENTS', true)
 	internal_conf.set10('MLIBC_MAP_FILE_WINDOWS', true)
 	subdir('sysdeps/astral')
+elif host_machine.system() == 'salernos'
+	rtld_include_dirs += include_directories('sysdeps/salernos/include')
+	libc_include_dirs += include_directories('sysdeps/salernos/include')
+	subdir('sysdeps/salernos')
 elif host_machine.system() == 'menix'
 	rtld_include_dirs += include_directories('sysdeps/menix/include')
 	libc_include_dirs += include_directories('sysdeps/menix/include')
